src/KingSystem/Map/mapObject.cpp

```diff
@@ -1,7 +1,373 @@
+
+    if (mProc->isInit() || mProc->isCalc())
+        return accessor.acquire(mProc);
+
+    return accessor.acquire(nullptr);
+}
+
+act::Actor* Object::getActor() const {
+    auto* actor = sead::DynamicCast<act::Actor>(mProc);
+    if (actor == nullptr)
+        return nullptr;
+    return actor;
+}
+
+void Object::registerBaseProc(act::BaseProc* proc) {
+    const char* sActorStateStrings[4] = {"Init", "Calc", "Sleep", "Delete"};
+
+    if (mProc != nullptr && mProc != proc) {
+        sead::FixedSafeString<1000> message;
+        message.format("%p   (%s:%p:%s) → (%s:%p:%s) \n", this, mProc->getName().cstr(), mProc,
+                       sActorStateStrings[static_cast<s32>(mProc->getState())],
+                       proc->getName().cstr(), proc,
+                       sActorStateStrings[static_cast<s32>(proc->getState())]);
+        util::PrintDebug(message);
+    }
+    mProc = proc;
+
+    mFlags0.set(Flag0::_80);
+    mFlags0.reset(Flag0::_80000000);
+    mHardModeFlags.set(HardModeFlag::ActorCreated);
+}
+
+void Object::setBaseProcDirect(act::BaseProc* proc) {
+    mProc = proc;
+}
+
+bool Object::isEnemyOrNpcOrActiveOrMapPassive() const {
+    if (mActorFlags8.isOn(ActorFlag8::EnemyOrNpcOrActiveOrAreaOrAirWall))
+        return true;
+
+    auto pa = PlacementMgr::instance()->mPlacementActors;
+    if (pa == nullptr)
+        return false;
+
+    return pa->mActorData[mActorDataIdx].mFlags.isOffBit(ActorData::Flag::MapPassiveOrFlag1);
+}
+
+bool Object::isFlags8Cleared() const {
+    return mActorFlags8.isZero();
+}
+
+bool Object::isNpcOrActiveOrMapPassiveOrFlag1() const {
+    auto actFlags = mActorFlags8;
+    if (actFlags.isZero())
+        return true;
+
+    auto pa = PlacementMgr::instance()->mPlacementActors;
+    if (pa == nullptr)
+        return false;
+
+    if (pa->mActorData[mActorDataIdx].mFlags.isOnBit(ActorData::Flag::_16) ||
+        actFlags.isOn(ActorFlag8::EnemyOrNpcOrActiveOrAreaOrAirWall))
+        return false;
+
+    return pa->mActorData[mActorDataIdx].mFlags.isOnBit(ActorData::Flag::MapPassiveOrFlag1);
+}
+
+bool Object::isMapPassive(const PlacementActors* pa) const {
+    if (mActorFlags8.isZero())
+        return true;
+
+    if (pa == nullptr)
+        return false;
+
+    if (pa->mActorData[mActorDataIdx].mFlags.isOnBit(ActorData::Flag::_16) ||
+        mActorFlags8.isOn(ActorFlag8::EnemyOrNpcOrActiveOrAreaOrAirWall))
+        return false;
+
+    return pa->mActorData[mActorDataIdx].mFlags.isOnBit(ActorData::Flag::MapPassiveOrFlag1);
+}
+
+bool Object::checkActorDataFlag(const PlacementActors* pa, ActorData::Flag flag) const {
+    return pa->mActorData[mActorDataIdx].mFlags.isOnBit(flag);
+}
+
+bool Object::isEnemyOrNpc(const PlacementActors* pa) const {
+    return checkActorDataFlag(pa, ActorData::Flag::EnemyOrNpc_DisableFlashback);
+}
+
+f32 Object::getDispDistance(const ActorData* data, bool getDiameter, u32 unused,
+                            bool ignoreRadius) const {
```

Comment:
```suggestion
                            bool ignore_radius) const {
```

N uses snake_case for local variables